pages/docs/tutorials/native/apple-framework.md

```diff
@@ -245,34 +269,76 @@ into Swift. As the result we'll have the following code in `main.swift`:
-the only one instance. Kotlin `object DemoObject` is turned to have a
-constructor in Swift and we use the `DemoDemoObject()` syntax to access it.
-All the time we have the same instance in Swift, so that 
-`DemoDemoObject() === DemoDemoObject()` is true.
-We may see from the Objective-C attributes 
-in `<Framework>/Headers/Demo.h` that it is the way to access the object instance. 
+the only one instance. Kotlin `object Object` is turned to have a
+constructor in Swift and we use the `DemoObject()` syntax to access the only instance of it.
+The instance is always the same in Swift, so that 
+`DemoDemoObject() === DemoDemoObject()` is true. 
 Methods and property names are translated as-is. Kotlin `String` is turned into Swift `String` too.
+Swift hides `NSNumber*` boxing from us too. We pass Swift closure to Kotlin and are able
+to call Kotlin lambda function from Swift too. 
+
+You may want to see [Objective-C Interop](https://github.com/JetBrains/kotlin-native/blob/master/OBJC_INTEROP.md)
+documentation article to learn all other types mapping details in detail.
 
 # XCode and Framework Dependencies
 
-It may turn out tricky to configure the Objective-C or Swift project in XCode or
-JetBrains AppCode. You should include the Framework into the *General* tab of Target settings
-of XCode project. Also, you may need to fix the `@rpath` to include our Framework folder in the 
-search path, for that
-just search for *Runpath Search Paths* in the *Build Settings* section of a target. 
+We need to configure XCode project to use out Framework. The configuration depends on the
+target platform. 
+
+## XCode for MacOS Target
+
+First we need to include the Framework in the `General` section of the *target*
+configuration. There is the `Linked Frameworks and Libraries` section to include
+out Framework. That will make XCode see our Framework and resolve imports both
+from Objective-C and Swift.
+
+The second step is to configure the Framework search path of the produced
+binary. It is also known as `rpath` or [run-time search path](https://en.wikipedia.org/wiki/Rpath).
+The binary uses the path to look for the required Frameworks. We do not recommend
+installing you frameworks to the OS without the need. You should understand the layout
+of your future application, for example, 
+you may have `Frameworks` folder under the application bundle with all frameworks you use. 
+The `@rpath` parameter can be configured in the XCode. You need to open
+*project* configuration and find the `Runpath Search Paths` section. There we specify
+the relative path to the compiled framework.
+
+You may also want to set XCode to create all project build files under the project
+root. It is done via `File | Project Settings` menu. That will simplify the way you
+pass the `rpath` to the executable. 
+
+## XCode for iOS Targets
+
+We add the Framework in the `Embedded Binaries` section of the `General` section of
```

Comment:
Is `Embedded Binaries` feature available for macOS target?

Reply:
Nope, for bugger Mac you normally include frameworks with the application and configure `@rpath` appropriately